src/analysis/sccp.rs

```diff
@@ -14,10 +14,23 @@
 
 use std::collections::{HashMap, VecDeque};
 use std::u64;
+use petgraph::graph::NodeIndex;
 use middle::ssa::ssa_traits::{SSA, SSAMod};
-use middle::ssa::ssa_traits::NodeType;
+use middle::ssa::ssa_traits::{NodeData, NodeType, ValueInfo, ValueType};
 use middle::ssa::graph_traits::{Graph, ConditionInfo};
-use middle::ir::{MArity, MOpcode};
+use middle::ir::{MArity, MOpcode, WidthSpec, MAddress};
+
+macro_rules! node_data_from_g {
```

Comment:
Btw, imo, it would be better to put this macro into middle/ssa/ssastorage.rs, which will be commonly used in the whole project.